src/cryptsetup/cryptsetup-tokens/cryptsetup-token-util.c

```diff
@@ -0,0 +1,47 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include "cryptsetup-token-util.h"
+#include "string-util.h"
+
+int crypt_dump_buffer_to_hex_string(
+                const char *buf,
+                size_t buf_size,
+                char **ret_dump_str) {
+
+        int r;
+
+        assert(ret_dump_str);
+
+        for (unsigned int i = 0; i < buf_size; i++) {
+                /* crypt_dump() breaks line after every
+                 * 16th couple of chars in dumped hexstring */
+                r = strextendf_with_separator(
+                        ret_dump_str,
+                        (i && !(i % 16)) ? CRYPT_DUMP_LINE_SEP : " ",
+                        "%02hhx", buf[i]);
+                if (r < 0)
+                        return r;
+        }
```

Comment:
we have hexdump() for cases like this, do we need another implementation of this?

Reply:
Both helpers are used solely in context of crypt_dump() output.

(example of cryptsetup luksDump output which invokes crypt_dump())

```
(...)
  2: luks2
        Key:        512 bits
        Priority:   normal
        Cipher:     aes-xts-plain64
        Cipher key: 512 bits
        PBKDF:      pbkdf2
        Hash:       sha512
        Iterations: 1000
        Salt:       10 ce c5 6a ec 78 d1 d3 b1 4f 80 5f a3 c4 2a 7e 
                    b9 55 2a 34 b4 81 93 80 c6 2f 91 95 13 c7 8f a6 
        AF stripes: 4000
        AF hash:    sha512
        Area offset:548864 [bytes]
        Area length:258048 [bytes]
        Digest ID:  0
Tokens:
  0: systemd-tpm2
        tpm2-pcrs:  7
        tmp2-blob:  00 9e 00 20 36 36 1c 2d 75 7a 19 95 0d e5 c8 78
                    d8 7e 1e 07 18 2b 6f 36 82 3f 0a 29 c8 00 fe 5a
(...)
```

I wanted it to indent and look same as example above. Hexdump() does some extra bits and indent differently IIUC. Or can is there any tunable version of it?